dev/MRTCore/mrt/Microsoft.Windows.ApplicationModel.Resources/src/ResourceContext.cpp

```diff
@@ -1,15 +1,76 @@
-﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Copyright (c) Microsoft Corporation and Contributors.
 // Licensed under the MIT License.
 
 #include "pch.h"
 #include "ResourceContext.h"
 #include "ResourceContext.g.cpp"
 #include "winrt/Windows.Globalization.h"
+#include <wil\resource.h>
 
 const wchar_t c_languageQualifierName[] = L"Language";
 
 namespace winrt::Microsoft::Windows::ApplicationModel::Resources::implementation
 {
+
+ResourceContext::~ResourceContext()
+{
+    winrt::slim_lock_guard guard {m_lock};
+    MrmDestroyResourceContext(m_resourceContext);
+}
+
+
+winrt::Windows::Foundation::Collections::IMap<hstring, hstring> ResourceContext::QualifierValues()
+{
+    m_lock.lock_shared();
+
+    auto cleanupOnFailure = wil::scope_exit([&] { m_lock.unlock_shared(); });
```

Comment:
FWIW, STL has https://en.cppreference.com/w/cpp/thread/call_once that makes this pattern easier.

There's no issue with WIL, of course. Other components in this repo use it just fine.

Reply:
Done, now using call_once, thanks